src/main/java/org/codebusters/audiogeek/preferencesagent/application/rest/mygenres/MyGenresRestAdapter.java

```diff
@@ -23,20 +23,33 @@ class MyGenresRestAdapter {
     private final PutMyGenresPort putMyGenresPort;
     private final GenreConverter genreConverter;
 
-    //TODO use correct ID after integration with Spring Security
-    private static final UserID DUMMY_ID = new UserID(UUID.fromString("45ca6b95-6cf3-4aca-b651-84a2a2aa8cd7"));
-
+    /**
+     * Function handling /my-genres GET request
+     * @param token token authorizing user
+     * @param authentication token payload via Spring Security
+     * @return 200 OK and GetMyGenresResponse
+     */
     @GetMapping(value = "/my-genres", produces = APPLICATION_JSON_VALUE, consumes = APPLICATION_JSON_VALUE)
-    public ResponseEntity<GetMyGenresResponse> getMyGenres(@RequestHeader("Authorization") String token) {
+    public ResponseEntity<GetMyGenresResponse> getMyGenres(@RequestHeader("Authorization") String token, Authentication authentication) {
         log.info("Processing GET request /my-genres");
-        return ResponseEntity.ok(new GetMyGenresResponse(genreConverter.toString(getMyGenresPort.getMyGenres(DUMMY_ID))));
+        var payload = (HuellTokenPayload) authentication.getPrincipal();
+        return ResponseEntity.ok(new GetMyGenresResponse(genreConverter.toString(getMyGenresPort.getMyGenres(new UserID(payload.id())))));
     }
 
+    /**
+     * Fuction handling /my-genres PUT request
+     * @param request request data (PutMyGenresRequest)
+     * @param token token authorizing user
+     * @param authentication token payload via Spring Security
+     * @return 201 CREATED (set genres for user)
+     */
     @PutMapping(value = "/my-genres", produces = APPLICATION_JSON_VALUE, consumes = APPLICATION_JSON_VALUE)
     public ResponseEntity<Void> putMyGenres(@RequestBody PutMyGenresRequest request,
-                                            @RequestHeader("Authorization") String token) {
+                                            @RequestHeader("Authorization") String token,
+                                            Authentication authentication) {
+        var payload = (HuellTokenPayload) authentication.getPrincipal();
```

Comment:
Duplicated code. Why no add in auth package some Utility class with static method to extract UseID from Authentication object?